File: pkg/java/src/main/java/dev/openfga/language/errors/ValidationError.java
package dev.openfga.language.errors;

import com.fasterxml.jackson.annotation.JsonValue;

public enum ValidationError {
    AllowedTypesNotValidOnSchema1_0("allowed-type-not-valid-on-schema-1_0"),
    AssignableRelationsMustHaveType("assignable-relation-must-have-type"),
    ConditionNotDefined("condition-not-defined"),
    ConditionNotUsed("condition-not-used"),
    DuplicatedError("duplicated-error"),
    InvalidName("invalid-name"),
    InvalidRelationType("invalid-relation-type"),
    InvalidSchema("invalid-schema"),
    InvalidSyntax("invalid-syntax"),
    InvalidType("invalid-type"),
    MissingDefinition("missing-definition"),
    RelationNoEntrypoint("relation-no-entry-point"),
    RequireSchema1_0("allowed-type-schema-10"),
    ReservedRelationKeywords("reserved-relation-keywords"),
    ReservedTypeKeywords("reserved-type-keywords"),
    SchemaVersionRequired("schema-version-required"),
    SchemaVersionUnsupported("schema-version-unsupported"),
    SelfError("self-error"),
    TuplesetNotDirect("tupleuserset-not-direct"),
    TypeRestrictionCannotHaveWildcardAndRelation("type-wildcard-relation"),
    InvalidRelationOnTupleset("invalid-relation-on-tupleset"),
    DifferentNestedConditionName("different-nested-condition-name"),
    MultipleModulesInFile("multiple-modules-in-file");

    private final String value;

    ValidationError(String value) {
        this.value = value;
    }

    @JsonValue
    public String getValue() {
        return value;
    }
}

File: pkg/csharp/src/OpenFga.Language/Validation/ValidationError.cs
using System.Text.Json.Serialization;

namespace OpenFga.Language.Validation;

public enum ValidationError
{
    [JsonPropertyName("allowed-type-not-valid-on-schema-1_0")]
    AllowedTypesNotValidOnSchema1_0,
    
    [JsonPropertyName("assignable-relation-must-have-type")]
    AssignableRelationsMustHaveType,
    
    [JsonPropertyName("condition-not-defined")]
    ConditionNotDefined,
    
    [JsonPropertyName("condition-not-used")]
    ConditionNotUsed,
    
    [JsonPropertyName("duplicated-error")]
    DuplicatedError,
    
    [JsonPropertyName("invalid-name")]
    InvalidName,
    
    [JsonPropertyName("invalid-relation-type")]
    InvalidRelationType,
    
    [JsonPropertyName("invalid-schema")]
    InvalidSchema,
    
    [JsonPropertyName("invalid-syntax")]
    InvalidSyntax,
    
    [JsonPropertyName("invalid-type")]
    InvalidType,
    
    [JsonPropertyName("missing-definition")]
    MissingDefinition,
    
    [JsonPropertyName("relation-no-entry-point")]
    RelationNoEntrypoint,
    
    [JsonPropertyName("allowed-type-schema-10")]
    RequireSchema1_0,
    
    [JsonPropertyName("reserved-relation-keywords")]
    ReservedRelationKeywords,
    
    [JsonPropertyName("reserved-type-keywords")]
    ReservedTypeKeywords,
    
    [JsonPropertyName("schema-version-required")]
    SchemaVersionRequired,
    
    [JsonPropertyName("schema-version-unsupported")]
    SchemaVersionUnsupported,
    
    [JsonPropertyName("self-error")]
    SelfError,
    
    [JsonPropertyName("tupleuserset-not-direct")]
    TuplesetNotDirect,
    
    [JsonPropertyName("type-wildcard-relation")]
    TypeRestrictionCannotHaveWildcardAndRelation,
    
    [JsonPropertyName("invalid-relation-on-tupleset")]
    InvalidRelationOnTupleset,
    
    [JsonPropertyName("different-nested-condition-name")]
    DifferentNestedConditionName,
    
    [JsonPropertyName("multiple-modules-in-file")]
    MultipleModulesInFile
}
